### ISSUE TITLE
Not all usage issues reported back to user

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#1714]
``` r
tst <- function() { a <- ...; a; b }
codetools::checkUsage(tst)
#> <anonymous>: ... may be used in an incorrect context (<text>:1)
#> <anonymous>: no visible binding for global variable 'b' (<text>:1)
```

<sup>Created on 2022-10-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Lintr only reports one of the problems:

``` r
lintr::lint(text = "tst <- function() { a <- ...; a; b }", linters = lintr::object_usage_linter())
#> <text>:1:34: warning: [object_usage_linter] no visible binding for global variable 'b'
#> tst <- function() { a <- ...; a; b }
#>                                  ^
```

<sup>Created on 2022-10-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

___

Checking the code of `R/codetools.R`, all issues are collected using the format `signalUsageIssue(message)`. The regex - if I read it correctly - implicitly assumes there always is an `sQuote(...)` in the message.

This is not the case for the following messages, scouring the source code:

```
w$signal("... may be used in an incorrect context", w) # l. 416
w$signal(paste(a, "may be used in an incorrect context:", pasteExpr(e)), w) # l. 436
w$signal(paste("wrong number of arguments to '.Internal':", pasteExpr(e)), w) # l. 676
w$signal(paste("bad argument to '.Internal':", pasteExpr(e[[2]])), w) # l. 682
```

_Originally posted by @AshesITR in https://github.com/r-lib/lintr/pull/1661#pullrequestreview-1143188740_

**Comment by AshesITR** [comment]
Here's a quick and dirty fix that will also make the `if (any(missing))` conditional obsolete:

```r
  res <- re_matches(
    vals,
    rex(
      function_name,
      capture(
        name = "message",
        zero_or_more(any, type = "lazy"),
        maybe(
          one_of(quote, "\u2018"),
          capture(name = "name", anything),
          one_of(quote, "\u2019"),
          anything
        )
      ),
      line_info
    )
  )
```
